pkg/controllers/provisioner/v1alpha1/controller.go

```diff
@@ -61,17 +64,48 @@ func (c *Controller) Reconcile(object controllers.Object) error {
 
 	unschedulable := []*v1.Pod{}
 	for _, pod := range pods.Items {
+		tempPod := pod
```

Comment:
Why take this reference instead of appending pod directly?

Reply:
We add a pointer reference in the append call, this is one of the big Go gotchas.
https://play.golang.org/p/6vSFa4QWN6d